ru/basecall.py

```diff
@@ -193,3 +191,30 @@ def map_reads_2(self, calls):
         """
         for read_info, read_id, seq, seq_len, quality in calls:
             yield read_info, read_id, seq_len, list(self.mapper.map(seq))
+
+
+class MappyRSMapper:
+    def __init__(self, index: Union[Path, str]) -> None:
+        self.index = index
+        if self.index:
+            self.mapper = mp.Aligner(6, self.index)
+            self.initialised = True
+        else:
+            self.mapper = None
+            self.initialised = False
+
+    def map_batch(self, iterable: Iterable[tuple[tuple[int, str], dict[Any]]]):
+        """
+        Consume an iterable sending it to the mapper and pulling back results
+        """
+        cache = {}
+        # count, got = 0, 0
+        for id, metadata in iterable:
+            cache[id] = metadata
+            res = self.mapper.send_one((id, metadata["datasets"]["sequence"]))
```

Comment:
Probably safe to use `metadata[...]` to get the values, maybe consider using `metadata.get(...)`. Though if this goes wrong so has `basecall.py`

Reply:
Switched to .get with sensible defaults